### PR TITLE
error on compile fail

### PR DESCRIPTION
**Pull request by nathanrpage97**
Implementation of #10 issue

### CODE REVIEW

File: jsregexp.c [hunk 1]
```diff
@@ -310,9 +309,28 @@ static int jsregexp_compile(lua_State *lstate)
   return 1;
 }
 
+static int jsregexp_compile_safe(lua_State *lstate) {
+  luaL_checkstring(lstate, 1);
+  luaL_optstring(lstate, 2, NULL);
+  lua_pushcfunction(lstate, jsregexp_compile);
+  lua_pushvalue(lstate, 1);
+  lua_pushvalue(lstate, 2);
+  int err = lua_pcall(lstate, 2, 1, 0);
```
Comment (kmarius) [inline jsregexp.c]: This looks a bit convoluted, I think one can
1. get the number of passed arguments
2. push `jsregex_compile` and insert it at stack position 1
3. call `lua_pcall` with the number of arguments

And on failure, push `nil` and insert it below the error.

File: jsregexp.c [hunk 2]
```diff
@@ -279,9 +279,9 @@ static int jsregexp_compile(lua_State *lstate)
 
   // lre_compile can segfault if the input contains 0x8f, which
   // indicated the beginning of a six byte sequence, but is now illegal.
+  
   if (strchr(regexp, 0xfd)) {
-    lua_pushnil(lstate);
-    lua_pushstring(lstate, "malformed unicode");
+    luaL_argerror(lstate, 1, "malformed unicode");
     return 2;
```
Comment (kmarius) [inline jsregexp.c]: Not sure if it should be a `luaL_argerror` or simply a `luaL_error`. I have no strong opinion on it. Anyway, the `return` can go.

**Comment by nathanrpage97** [comment]
ok I've cleaned up the commits and git diff. It is ready for a review.

File: jsregexp.c (hunk 1 above)
Comment (kmarius) [reply jsregexp.c]: I'd still like to see this cleaner. Push `jsregexp_compile`, insert at position `1` call it like

``` lua
if (lua_pcall(lstate, lua_gettop(lstate)-1, 1, 0) != 0) {
    ...
}
```
No need to push the values again.

**Review by kmarius — COMMENTED** [review]
Ok, I think I was using the code review wrong. `compile_safe` needs some tidying and we are good to go.

**Comment by kmarius** [comment]
Thanks!

**Pull request merged**